### PR TITLE
test: Code coverage setupAccountPage 

### PR DESCRIPTION
**Pull request by CODEX108**
clickUp: https://app.clickup.com/t/85zt051zw

### CODE REVIEW

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,300 @@
...
+    expect(component.setupNetworkWatcher).toHaveBeenCalledTimes(1);
+    expect(orgService.setCurrencyBasedOnIp).toHaveBeenCalledTimes(2);
+    expect(orgService.getCurrentOrg).toHaveBeenCalledTimes(2);
+    component.org$.subscribe(() => {
+      expect(component.fg.controls.homeCurrency.value).toBe(org.currency);
+    });
+    tick(500);
+  }));
+
+  xit('should emit the correct length validation display value when the password value changes', fakeAsync(() => {
```
Comment (CODEX108) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:156]: The spec passes without expectations(when I try checking single input at a time / throws error stating `expected false to be true` when I try looping around the assertion.

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Draft pr with commented code. Please check pr once before assigning

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 2]
```diff
@@ -1,26 +1,348 @@
...
+    const expectedOrgData = {
+      ...orgData1[0],
+      name: 'Uber',
+      currency: 'USD',
+    };
+    component.postOrg().subscribe(() => {
+      expect(orgData1[0]).toEqual(expectedOrgData);
+      expect(orgService.updateOrg).toHaveBeenCalledOnceWith(expectedOrgData);
+    });
+    tick(500);
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:170]: The test should work without the tick as well

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 3]
```diff
@@ -1,26 +1,348 @@
...
+  }));
+
+  describe('saveGuessedMileage():', () => {
+    it('should set the desired mileage value if the org currency is USD', fakeAsync(() => {
+      orgSettingsService.get.and.returnValue(of(orgSettingsRes));
+      component.org$ = of(orgData1[0]);
+      orgSettingsService.post.and.returnValue(of(orgSettingsPostData));
+      component.saveGuessedMileage().subscribe(() => {
+        fixture.detectChanges();
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:179]: Should not be needed

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 4]
```diff
@@ -1,26 +1,348 @@
...
+        expect(orgSettingsRes.mileage.four_wheeler).toBe(0.58);
+        expect(orgSettingsRes.mileage.unit).toBe('MILES');
+        expect(orgSettingsRes.mileage.enabled).toBe(true);
+        expect(orgSettingsRes.mileage.two_wheeler).toBe(0.58);
+      });
+      tick(500);
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:185]: same as above

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 5]
```diff
@@ -1,26 +1,348 @@
...
+    it('onInit(): should set the homeCurrency control value from org$', fakeAsync(() => {
+      spyOn(component, 'setupNetworkWatcher');
+      const org = { currency: 'USD' };
+      orgService.setCurrencyBasedOnIp.and.returnValue(of(orgData1[0]));
+      orgService.getCurrentOrg.and.returnValue(of(orgData1[0]));
+      component.ngOnInit();
+      fixture.detectChanges();
+      tick(500);
+      expect(component.setupNetworkWatcher).toHaveBeenCalledTimes(1);
+      expect(orgService.setCurrencyBasedOnIp).toHaveBeenCalledTimes(2);
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:273]: Should be called only once

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 6]
```diff
@@ -1,26 +1,348 @@
...
+      expect(orgService.getCurrentOrg).toHaveBeenCalledTimes(2);
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:274]: Same as above

Comment (yksolanki9) [reply src/app/post-verification/setup-account/setup-account.page.spec.ts:274]: Removing the explicit call to `component.ngOnInit()` should fix this

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 7]
```diff
@@ -1,26 +1,348 @@
...
+  describe('onInit()', () => {
+    it('onInit(): should set the homeCurrency control value from org$', fakeAsync(() => {
+      spyOn(component, 'setupNetworkWatcher');
+      const org = { currency: 'USD' };
+      orgService.setCurrencyBasedOnIp.and.returnValue(of(orgData1[0]));
+      orgService.getCurrentOrg.and.returnValue(of(orgData1[0]));
+      component.ngOnInit();
+      fixture.detectChanges();
+      tick(500);
+      expect(component.setupNetworkWatcher).toHaveBeenCalledTimes(1);
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:272]: Add assertions to check if eou$ and fullname$ are set

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 8]
```diff
@@ -1,26 +1,348 @@
...
+      ];
+      testCases.forEach((testCase) => {
+        component.uppercaseValidationDisplay$.pipe(take(1)).subscribe((value) => {
+          expect(value).toEqual(testCase.expectedOutput);
+        });
+        component.fg.controls.password.patchValue(testCase.input);
+      });
+    });
+
+    it('should emit the correct value to check for number vadility', () => {
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:309]: validity

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 9]
```diff
@@ -1,26 +1,348 @@
...
+      ];
+      testCases.forEach((testCase) => {
+        component.numberValidationDisplay$.pipe(take(1)).subscribe((value) => {
+          expect(value).toEqual(testCase.expectedOutput);
+        });
+        component.fg.controls.password.patchValue(testCase.input);
+      });
+    });
+
+    it('should emit the correct value to check for lowerCase vadility', () => {
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:322]: Fix the spelling of validity in every test case

File: src/app/post-verification/setup-account/setup-account.page.spec.ts [hunk 10]
```diff
@@ -1,26 +1,348 @@
...
+      ];
+      testCases.forEach((testCase) => {
+        component.lowercaseValidationDisplay$.pipe(take(1)).subscribe((value) => {
+          expect(value).toEqual(testCase.expectedOutput);
+        });
+        component.fg.controls.password.patchValue(testCase.input);
+      });
+    });
+
+    it('should emit the correct value to check for lowerCase vadility', () => {
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/setup-account.page.spec.ts:335]: Fix the spec name

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Minor comments

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**